File: annAli.py
# -*- coding: utf-8 -*-
"""
Created on Wed Dec  6 04:09:24 2017

@author: Ali
"""
#### PART I DATA PREPROCESSING \

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Churn_Modelling.csv')
X = dataset.iloc[:, 3:13].values
y = dataset.iloc[:, 13].values

# Encoding categorical data
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X_1 = LabelEncoder()
X[:, 1] = labelencoder_X_1.fit_transform(X[:, 1])
labelencoder_X_2 = LabelEncoder()
X[:, 2] = labelencoder_X_2.fit_transform(X[:, 2])
onehotencoder = OneHotEncoder(categorical_features = [1])
X = onehotencoder.fit_transform(X).toarray()
X = X[:, 1:] #Preventing the Dummy Variable Trap 

# Splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

### PART II MAKING THE ANN MODEL 
#Importing the Keras libraries and packages
import keras
from keras.models import Sequential
from keras.layers import Dense
#Initializing the ANN 
classifier = Sequential()
#Adding the input layer and the first hidden layer 
classifier.add(Dense(units = 6, kernel_initializer = 'uniform', activation = 'relu',input_dim = 11))
#Adding the second hidden layer 
classifier.add(Dense(units = 6, kernel_initializer = 'uniform', activation = 'relu'))
#Adding the output layer 
classifier.add(Dense(units = 1, kernel_initializer = 'uniform', activation = 'sigmoid')) #if output = 3 then activation = softmax
#Compiling the ANN 
classifier.compile(optimizer = 'adam', loss = 'binary_crossentropy', metrics =['accuracy'])
#Fitting the ANN to the Training Set 
classifier.fit(X_train, y_train, batch_size = 20, epochs =100 )

#PART III MAKING THE PREDICTIONS AND EVALUATING THE MODEL 
# Predicting the Test set results
y_pred = classifier.predict(X_test)
y_pred = (y_pred > 0.5) #If y_pred larger than 0.5 then true, otherwise false 
# Making the Confusion Matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)

File: annAli.R

# Importing the dataset
dataset = read.csv('Churn_Modelling.csv')
dataset = dataset[4:14]

# Encoding the target feature as factor
dataset$Geography = as.numeric(factor(dataset$Geography, levels = c('France', "Spain", "Germany"), labels = c(1,2,3))) #DON'T fORGET TO CHANGE IT INTO NUMERIC

dataset$Gender = as.numeric(factor(dataset$Gender, levels = c('Female', "Male"), labels=c(0,1)))

# Splitting the dataset into the Training set and Test set
library(caTools)
set.seed(123)
split = sample.split(dataset$Exited, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling
training_set[-11] = scale(training_set[-11])
test_set[-11] = scale(test_set[-11])

# Fitting ANN to the Training set
#install.packages("h2o")
library(h2o)
h2o.init(nthreads = -1)
classifier=h2o.deeplearning(y = 'Exited', 
                            training_frame = as.h2o(training_set), #Turning into an h2o frame
                            activation = 'Rectifier',
                            hidden = c(6,6),
                            epochs = 100,
                            train_samples_per_iteration = -2)


# Predicting the Test set results
prob_pred = h2o.predict(classifier, newdata=as.h2o(test_set[-11]))
y_pred = (prob_pred>0.5) #BOOLEAN, But still an h2o object
y_pred= as.vector(y_pred) #changing it again into vector

# Making the Confusion Matrix
cm = table(test_set[, 11], y_pred)

h2o.shutdown()
